python_modules/dagster/dagster/_core/definitions/partitions/context.py

```diff
@@ -14,6 +14,9 @@
 if TYPE_CHECKING:
     from dagster._core.instance import DynamicPartitionsStore
 
+# Type variable for preserving return type
+T = TypeVar("T")
```

Comment:
this stuff is unrelated, just makes type hints work better in this part of the code base